pyhf/constraints.py

```diff
@@ -0,0 +1,87 @@
+from . import get_backend
+
+class gaussian_constraint_combined(object):
+    def __init__(self,pdf):
+        tensorlib, _ = get_backend()
+        # iterate over all constraints order doesn't matter....
+        start_index = 0
+        summands = None
+        
+        normal_constraint_data = []
+        normal_constraint_mean_indices = []
+        normal_constraint_sigmas = []
+        
+        par_indices = list(range(len(pdf.config.suggested_init())))
+        data_indices = list(range(len(pdf.config.auxdata)))
+        for cname in pdf.config.auxdata_order:
+            modifier = pdf.config.modifier(cname)
+            if not modifier.pdf_type == 'normal': continue
+            modslice  = pdf.config.par_slice(cname)
+
+            end_index = start_index + modifier.n_parameters
+            thisauxdata = data_indices[start_index:end_index]
+            start_index = end_index
+
+            normal_constraint_sigmas.append(modifier.sigmas)
+            normal_constraint_data.append(thisauxdata)
+            normal_constraint_mean_indices.append(par_indices[modslice])
+        
+        if normal_constraint_mean_indices:
+            normal_mean_idc  = tensorlib.concatenate(map(lambda x: tensorlib.astensor(x,dtype = 'int'),normal_constraint_mean_indices))
+            normal_sigmas    = tensorlib.concatenate(map(tensorlib.astensor,normal_constraint_sigmas))
+            normal_data      = tensorlib.concatenate(map(lambda x: tensorlib.astensor(x,dtype = 'int'),normal_constraint_data))
+        else:
+            normal_data, normal_sigmas, normal_mean_idc = None, None, None
+
+        self.prepped =  (normal_data,normal_sigmas,normal_mean_idc)
```

Comment:
replace with `self.normal_data, self.normal_sigmas, self.normal_mean_idc` here. `self.prepped` is not very useful on its own and vague. Should also be called more like `normal_data_idx`, `normal_sigma_idx`, and `normal_mean_idx` or something.

Reply:
good point